Check if publicName already includes .exe

ulrikstrid opened the pull request:
This PR fixes the double .exe issue we had before on Windows.

ulrikstrid commented:
A interesting side-effect from this is that I now hit the same issue as I do in `hello-reason` and other cross-platform release attempts:
```
PS C:\LocalDevelopment\reenv\_release> reenv.exe
Program 'reenv.exe' failed to run: The specified executable is not a valid application for this OS
platform.At line:1 char:1
+ reenv.exe
+ ~~~~~~~~~.
At line:1 char:1
+ reenv.exe
+ ~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (:) [], ApplicationFailedException
    + FullyQualifiedErrorId : NativeCommandFailed
```
If I instead run the command as `reenv.exe.cmd` it works correctly, I'm not sure if this is something `npm` does or if it's on our end.

andreypopp commented on bin/NpmReleaseCommand.re:
```diff
@@ -623,7 +623,13 @@ let make =
-                    | Windows => publicName ++ ".exe"
+                    | Windows =>
+                      let publicNameLength = String.length(publicName);
+                      if (String.sub(publicName, publicNameLength - 4, 4) == ".exe") {
```
I think this fail if length of `publicName` is less than 4. Maybe just wrap it in `Path.t` and use its functions to check/add extension.

ulrikstrid commented:
I addressed your comments. I think this fix is good, but it's still not enough to get it working properly on Windows for some reason, at least not when `.exe` is added explicitly.

andreypopp commented:
Thanks!
